sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDD.scala

```diff
@@ -71,8 +72,15 @@ class StateStoreRDD[T: ClassTag, U: ClassTag](
       StateStoreId(checkpointLocation, operatorId, partition.index),
       queryRunId)
 
+    // If we're in continuous processing mode, we should get the store version for the current
+    // epoch rather than the one at planning time.
+    val currentVersion = EpochTracker.getCurrentEpoch match {
+      case -1 => storeVersion
```

Comment:
I dont like the fact that -1 is used as value for non-existence, especially since this value is present deep inside the EpochTracker. Might as well use Options to return None. Then the use of -1 will be contained within the EpochTracker class.